### DISCUSSION TITLE
Why nested YAML aren't treated like YAML?

### DISCUSSION THREAD

**Discussion by rohanbatraio** [RafaelGB/obsidian-db-folder#513] [Ideas]
I was wondering to make a decent system for tracking everything. For that nested tags are the best options to clutter free my YAML. So i thought why not go with that, but dbfolder doesn't seem to work as beautifully as it works in tags form. What i was expecting is to enable me add nested tags in the same way as adding it in case of tags, so that the datasource would be different and UI remains the same. Here's what i mean

See the tags "as genre" here? 

https://user-images.githubusercontent.com/11924043/172692604-f6a2df6a-85da-4883-b25e-d80fef750142.mov

Same i was expecting with nested tags, how? 
Here's how -> adding a nested tags query as a column would solve the problem and would be awesome to have. Would really stress on this feature being possible here. 

![dbfolder](https://user-images.githubusercontent.com/113782197/196100578-fd92942e-84c2-49b7-a107-a88cbac086c8.png)

**Comment by rohanbatraio** [comment]
Why would we consider this ? 

Here's what my workflow implies :- 

For ex. lets say , I am doing academics, health, meetings etc things in a day. Now all that belongs to a day, which is a date, so its general to have all that info reside in a daily note. 

Now i would like to make 3 databases (dbfolder) , 1 for each. and add specific column in each of them. For elaboration : 

Academics would have academics related yaml
Health would have its columns and so on. 

But ultimately all of that would go into one single daily note, imagine having 100's of individual yaml entries just so that you can have all features of DBFOLDER and are forced not to use nested yaml. 

So for a clean approach i would request the solution i shared in the beginning of this discussion.

**Comment by RafaelGB** [comment]
to finally understand it. The tags would admit nesting under the nomenclature you share

<img width="1429" alt="Captura" src="https://user-images.githubusercontent.com/11924043/196106405-9e784990-93bd-4186-aa52-b37dc4be69f1.PNG">

The requirement is that the column has as id `tags`. Another option is to use inline mode and add # at the beginning of the word

**Reply by rohanbatraio** [reply]
And are we able to parse it as well?

To give an idea, here's what i mean.

we are able to add a YAML KEY as a column but are we able to add NESTED YAML key as a column and later be able to manipulate its VALUE?

Is that what we both are understanding?

**Reply by rohanbatraio** [reply]
Why? Because it would be cleaner. A lot cleaner

**Reply by rohanbatraio** [reply]
in the example you shared this is what i wanted 

Roadmap/pato : "Value"

Value manipulation (same as using a CHECKBOX to toggle true or false, or using a SELECT field to manipulate etc. etc.) 

This value manipulation is achieved with YAML KEYS as adding them as COLUMNS but not achievable with NESTED YAML KEYS as the column doesn't take it as an input parameter.

**Reply by RafaelGB** [reply]
since 2.7.2 you can use and edit nested fields.  The way is by editing the id in the following form: field.myNestedField

it is not designed for the way of including tags that you propose

**Reply by rohanbatraio** [reply]
> it is not designed for the way of including tags that you propose

 So what are your thoughts on the way i propose it?

**Reply by RafaelGB** [reply]
the problem with including / in ids is that it doesn't follow yaml standards and can break them.  The way it has been proposed in 2.7.2 is to create complex objects and make nesting more flexible

**Reply by rohanbatraio** [reply]
> The way it has been proposed in 2.7.2 is to create complex objects and make nesting more flexible

Yes i am updated with #442 

> the problem with including / in ids is that it doesn't follow yaml standards and can break them. 

Why not to follow YAML standards? Why not just go with ->

 `YAML :  `
  ` - NESTED_YAML : KEY `
  ` - NESTED_YAML2 : KEY2 `
  
   and we would parse it with 

yaml:nested-yaml  -> with double quotes if any programming syntax sanitation is required . instead of yaml.nested yaml.

**Reply by rohanbatraio** [reply]
or parsing could be done as described here -> 
https://user-images.githubusercontent.com/113782197/196100578-fd92942e-84c2-49b7-a107-a88cbac086c8.png

**Reply by rohanbatraio** [reply]
Is there a specific reason you are using field.myNestedField ? 

instead of field/myNestedField OR field:myNestedField

**Comment by RafaelGB** [comment]
```yaml
YAML :
- NESTED_YAML : KEY
- NESTED_YAML2 : KEY2
```
this format is readable for yaml, but currently the plugin does not have any types for lists of objects, only lists (tags)

Making a list of objects is a new and quite complex interface. I can't think of an easy way to implement it.

**Reply by rohanbatraio** [reply]
I am sorry i misunderstood a lot. Now it's clarified. 
 As indicated i used yaml in conjuction with column id as tags, but now it shows [object Object] in the ddbb table.

**Reply by rohanbatraio** [reply]
I mixed nested tags and nested yaml altogether. Goshh! being stressed out here because obsidian after 1.0.0 is breaking with everything i am doing, sorry if you had issues understanding me.

**Reply by RafaelGB** [reply]
could you share a screenshot?

Object object is shown when the cell expects a string and gets a Object (in this case a nested yaml)

**Reply by rohanbatraio** [reply]
https://user-images.githubusercontent.com/113782197/196180423-d1956b2e-02ad-4430-9c7d-2ad4729d5133.mp4

**Reply by rohanbatraio** [reply]
High possibility i did something wrong here!

**Reply by RafaelGB** [reply]
its not the way the nested yaml works.

You are reading the entire object selecting YAML and tags only accepts a list as an input

If you want to use a nested attribute as column, you have to rename the id first. In your example
```yaml
object:
  nestedAttribute:
    - A
    - B
    - C
```
Then in your column config rename the id with `object.nestedAttribute`

**Reply by rohanbatraio** [reply]
Thanks for correcting me again @RafaelGB , I think i would stay away from nested YAML and skip it entirely for use of inline fields. Far better choice for me.

**Reply by RafaelGB** [reply]
happy to help! Its the first iteration of nested fields, will be improved with the time
